classes/class-wc-connect-shipping-label.php

```diff
@@ -132,6 +132,42 @@ protected function get_states_map() {
 			return $result;
 		}
 
+		public function should_show_meta_box() {
+			global $post, $theorder;
+
+			if ( ! is_object( $post ) ) {
+				return FALSE;
+			}
+
+			if ( ! is_object( $theorder ) ) {
+				$theorder = wc_get_order( $post->ID );
+			}
+
+			if ( ! $theorder ) {
+				return FALSE;
+			}
+
+			// TODO: return TRUE if the order has already label meta-data
+
+			$base_location = wc_get_base_location();
+			if ( 'US' !== $base_location[ 'country' ] ) {
+				return FALSE;
+			}
+
+			$dest_address = $theorder->get_address( 'shipping' );
+			if ( $dest_address[ 'country' ] && 'US' !== $dest_address[ 'country' ] ) {
```

Comment:
Even in a newly created order, all the address fields are initialized with an empty string, so it will never be an `undefined index` (but it will be "falsey")